samples/unit-tests/stock-tools/indicators/demo.js

```diff
@@ -42,89 +42,605 @@ QUnit.test('Managing tech indicators in Stock Tools', function (assert) {
+    );
+    assert.close(
+        parseFloat(chart.yAxis[2].options.height),
+        20,
+        0.0001, // up to 0.0001% is fine
+        'Correct Height for Indicator axis after adding 1 indicator. Main axis originally 60% height'
+    );
+
+    assert.close(
+        parseFloat(chart.yAxis[2].options.top),
+        60,
+        0.0001, // up to 0.0001% is fine
+        'Correct Top for Indicator axis after adding 2 indicators. Main axis originally 100% height'
+    );
+
+    chart.navigationBindings.utils.manageIndicators.call(
+        chart.navigationBindings,
+        {
+            actionType: 'add',
+            linkedTo: 'aapl',
+            fields: {
+                'params.index': '0',
+                'params.period': '5'
+            },
+            type: 'atr'
+        }
+    );
+
+    assert.close(
+        parseFloat(chart.yAxis[0].options.height),
+        60,
+        0.0001, // up to 0.0001% is fine
+        'Correct Height for Main axis after adding 1 indicator. Main axis originally 60% height'
+    );
+    assert.close(
+        parseFloat(chart.yAxis[2].options.height),
+        20,
+        0.0001, // up to 0.0001% is fine
+        'Correct Height for Indicator axis after adding 1 indicator. Main axis originally 60% height'
+    );
+
+    assert.close(
+        parseFloat(chart.yAxis[2].options.top),
+        60,
+        0.0001, // up to 0.0001% is fine
+        'Correct Top for Indicator axis after adding 2 indicators. Main axis originally 100% height'
+    );
+    assert.close(
+        parseFloat(chart.yAxis[3].options.height),
+        20,
+        0.0001, // up to 0.0001% is fine
+        'Correct Height for Indicator axis after adding 1 indicator. Main axis originally 60% height'
+    );
+
+    assert.close(
+        parseFloat(chart.yAxis[3].options.top),
+        80,
+        0.0001, // up to 0.0001% is fine
+        'Correct Top for Indicator axis after adding 2 indicators. Main axis originally 100% height'
+    );
+
+    chart.navigationBindings.utils.manageIndicators.call(
+        chart.navigationBindings,
+        {
+            actionType: 'remove',
+            seriesId:
+                chart.series[chart.series.length - 2]
+                    .options.id
         }
+    );
+    assert.close(
+        parseFloat(chart.yAxis[0].options.height),
+        75,
+        0.0001, // up to 0.0001% is fine
+        'Correct Height for Main axis after deleting 1 indicator. Main axis originally 60% height'
+    );
+    assert.close(
+        parseFloat(chart.yAxis[2].options.height),
+        25,
+        0.0001, // up to 0.0001% is fine
+        'Correct Height for Indicator axis after deleting 1 indicator. Main axis originally 60% height'
+    );
+
+    assert.close(
+        parseFloat(chart.yAxis[2].options.top),
+        75,
+        0.0001, // up to 0.0001% is fine
+        'Correct Top for Indicator axis after deleting 1 indicator. Main axis originally 60% height'
+    );
+    chart.destroy();
```

Comment:
Another chart destroy. Perhaps we could just update items on the chart (e.g. yAxes)?

Reply:
I replaced the recreating of the chart with updates.